en/tutorials-and-examples/cms/authentication.rst

```diff
@@ -143,6 +143,158 @@ misleading links, or continue on to the next section. We won't be building out
 user editing, viewing or listing in this tutorial, but that is an exercise you
 can complete on your own.
 
-* Enabling Access Control
-* Updating Creation
-* Restricting Editing
+Restricting Article Access
+==========================
+
+Now that users can log in, we'll want to limit users to only edit articles that
+they created. We'll do this using an 'authorization' adapter. Since our
+requirements are pretty simple, we can write some simple code in our
+``ArticlesController``. But before we do that, we'll want to tell the
+``AuthComponent`` how our application is going to authorize actions. Update your
+``AppController`` adding the following::
+
+    public function isAuthorized($user)
+    {
+        // By default deny access.
+        return false;
+    }
+
+Next we'll tell ``AuthComponent`` that we want to use controller hook methods
+for authorization. Your ``AppController::initialize()`` method should now look
+like::
+
+        public function initialize()
+        {
+            // Existing code
+
+            $this->loadComponent('Flash');
+            $this->loadComponent('Auth', [
+                // Added this line
+                'authorize'=> 'Controller',
+
+                'authenticate' => [
+                    'Form' => [
+                        'fields' => [
+                            'username' => 'email',
+                            'password' => 'password'
+                        ]
+                    ]
+                ],
+                'loginAction' => [
+                    'controller' => 'Users',
+                    'action' => 'login'
+                ],
+                 // If unauthorized, return them to page they were just on
+                'unauthorizedRedirect' => $this->referer()
```

Comment:
`$this->referer()` return 3 thing 
-`redirect` on url 
-`referer` that you mention it
- else return '/' 

so `loop` occurs sometimes